test/Integration/IntegrationTestBase.cs

```diff
@@ -259,11 +280,17 @@ private static string GetFunctionsCoreToolsPath()
             return funcPath;
         }
 
-        private void TestOutputHandler(object sender, DataReceivedEventArgs e)
+
+        private DataReceivedEventHandler GetTestOutputHandler(int processId)
         {
-            if (e != null && !string.IsNullOrEmpty(e.Data))
+            return TestOutputHandler;
+
+            void TestOutputHandler(object sender, DataReceivedEventArgs e)
             {
-                this.TestOutput.WriteLine(e.Data);
+                if (e != null && !string.IsNullOrEmpty(e.Data))
```

Comment:
`if (!string.IsNullOrEmpty(e?.Data))`

Reply:
I will be removing the check for null as I haven't ever seen a case when the event is set to null and none of the examples in Microsoft Docs ([1](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.errordatareceived?view=net-6.0#:~:text=String.IsNullOrEmpty(outLine.Data)), [2](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.errordatareceived?view=net-6.0#:~:text=String.IsNullOrEmpty(errLine.Data)), [3](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.outputdatareceived?view=net-6.0#:~:text=String.IsNullOrEmpty(e.Data))) check for it.